test/parallel/test-promise-hook-create-hook.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+// Flags: --expose-internals
+const common = require('../common');
+const assert = require('assert');
+const { createHook } = require('internal/promise_hook');
+
+let init;
+let initParent;
+let before;
+let after;
+let resolve;
+
+const stop = createHook({
+  init: common.mustCall((promise, parent) => {
+    init = promise;
+    initParent = parent;
+  }, 3),
+  before: common.mustCall((promise) => {
+    before = promise;
+  }, 2),
+  after: common.mustCall((promise) => {
+    after = promise;
+  }, 1),
+  resolve: common.mustCall((promise) => {
+    resolve = promise;
+  }, 2)
+});
+
+function assertState(expectedInit, expectedInitParent, expectedBefore,
+  expectedAfter, expectedResolve) {
+  assert.strictEqual(init, expectedInit);
+  assert.strictEqual(initParent, expectedInitParent);
+  assert.strictEqual(before, expectedBefore);
+  assert.strictEqual(after, expectedAfter);
+  assert.strictEqual(resolve, expectedResolve);
+  init = undefined;
+  initParent = undefined;
+  before = undefined;
+  after = undefined;
+  resolve = undefined;
+}
+
+const parent = Promise.resolve(1);
+assertState(parent, undefined, undefined, undefined, parent);
+
+const child = parent.then(() => {
+  assertState(undefined, undefined, child, undefined, undefined);
+});
+assertState(child, parent);
+  
+const grandChild = child.then(() => {
+  assertState(undefined, undefined, grandChild, child, child);
```

Comment:
the one or the other comment could be helpful here to help a reader to faster understand the flow.